MDANSE/Src/MDANSE/Mathematics/Signal.py

```diff
@@ -308,3 +318,932 @@ def get_spectrum(signal, window=None, timeStep=1.0, axis=0, fft="fft"):
+
+        :Returns:
+            #. FrequencyDomain: named tuple containing the x-axis (frequency range) and y-axis (amplitude) of the filter
+        """
+        return self._freq_response
+
+    @freq_response.setter
+    def freq_response(
+        self, params: Tuple[TransferFunction, FrequencyRangeMethod]
+    ) -> None:
+        """Calculates the frequency response of the filter from the filter's transfer function numerator and denominator coefficients.
+
+        :Parameters:
+            #. params (Tuple[TransferFunction, FrequencyRangeMethod]):
+            Tuple contains the following elements:
+                - the rational polynomial expression for the filter transfer function, in terms of its numerator and denominator coefficients
+                - the method by which to compute the frequency range for displaying the filter
+        """
+        expr, method = params
+        units = (
+            Filter.FrequencyUnits.CYCLIC
+            if Filter.Flags.DIGITAL_ONLY in self.flags
+            else Filter.FrequencyUnits.ANGULAR
+        )
+
+        if method is Filter.FrequencyRangeMethod.FFT:
+            # Compute frequency range using FFT
+            freq_range = self.frequency_range(
+                self.n_steps, self.sample_freq ** (-1), units=units
+            )
+        elif (
+            self.custom_freq_range.any()
+            and method is Filter.FrequencyRangeMethod.CUSTOM
+        ):
+            # Use custom frequency range (assumes frequencies are rad/ps)
+            freq_range = copy(self.custom_freq_range)
+
+            # Convert frequency range to cyclic frequencies if necessary
+            if units == Filter.FrequencyUnits.CYCLIC:
+                freq_range *= self._angular_to_cyclic
+        else:
+            RuntimeError(
+                f"Could not find supplied frequency range around which filter frequency response will be computed. \nPlease set the 'custom_freq_range' attribute on the instance of {self.__class__}"
+            )
+
+        # Compute filter response around frequencies given in range
+        response = self.compute_frequencies(
+            transfer_function=expr, range=np.abs(freq_range)
+        )
+        self._freq_response = self.FrequencyDomain(*response)
+
+    @classmethod
+    def frequency_resolution(cls, num_steps: float, timestep: float, units):
+        """Returns the frequency resolution of the trajectory given N fixed timesteps.
+        Analogous to the bin-width of an FFT of the trajectory.
+
+        :Parameters:
+            #. num_steps (float): Number of simulation timesteps
+            #. timestep (float): Simulation timestep in picoseconds
+            #. units (FrequencyUnit): Frequency unit type for conversion (i.e. CYCLIC=THz, ANGULAR=rad/ps)
+        :Returns:
+            #. float: Frequency resolution
+        """
+        bin_width = 1 / (num_steps * timestep)
+        if units == Filter.FrequencyUnits.ANGULAR:
```

Comment:
```suggestion
        if units == Filter.FrequencyUnits.ANGULAR:
```
Does this want to be `is`?